### PR TITLE
 Move the logic of checking the availability of batch execution to the executor

### PR DESCRIPTION
**Pull request by breeswish**
## What have you changed? (mandatory)

We have some really complicated checking logic in aggregation executors, so it is no longer suitable to put them together in one function. This PR moves them back to places near the executor itself. This makes it more cohesive.

Additionally, this PR also changes the checking logic from returning a bool to returning a Result, so that the reasons that batch cannot be used can be outputted by simply return the Error.

Extracted from #3898

- [x] Merge #4482 first

## What are the type of the changes? (mandatory)

- Engineering (engineering change which doesn't change any feature or fix any issue)

### CODE REVIEW

File: src/coprocessor/dag/handler.rs [hunk 1]
```diff
@@ -111,9 +111,16 @@ impl DAGRequestHandler {
...
 
-        let is_batch = enable_batch_if_possible
-            && !is_streaming
-            && super::builder::DAGBuilder::can_build_batch(req.get_executors());
+        let mut is_batch = false;
+        if enable_batch_if_possible && !is_streaming {
+            let build_batch_result =
+                super::builder::DAGBuilder::check_build_batch(req.get_executors());
+            if let Err(e) = build_batch_result {
+                debug!("Coprocessor request cannot be batched"; "reason" => %e);
```
Comment (rleungx) [inline src/coprocessor/dag/handler.rs]: I think the `debug` level may be not enough here.

Comment (breeswish) [reply src/coprocessor/dag/handler.rs]: Do you like it to be `info`?

Pros: We can see this log in production without any modifications
Cons: Maybe too many log output before we fully implemented batch execution (maybe before GA)

Comment (rleungx) [reply src/coprocessor/dag/handler.rs]: @AndreMouche What do you think?

Comment (AndreMouche) [reply src/coprocessor/dag/handler.rs]: I think we should use `debug` here.

**Pull request merged**